### ISSUE TITLE
`bad-open-mode` produces illegible message when passing `mode=None`

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#5733]
### Bug description

```python
open(fp, mode=None)
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
bad-open-mode:161:0:161:25::"""%s"" is not a valid mode for open.":UNDEFINED
```

### Expected behavior

Maybe take the `repr()` of things that aren't strings?

'None' is not a valid mode...

### Pylint version

```shell
pylint 2.13.0-dev0
astroid 2.9.3
Python 3.10.1 (v3.10.1:2cd268a3a9, Dec  6 2021, 14:28:59) [Clang 13.0.0 (clang-1300.0.29.3)]
```
